components/ui-api-layer/internal/domain/servicecatalog/serviceplan_converter.go

```diff
@@ -92,12 +98,28 @@ func (p *servicePlanConverter) unpackCreateParameterSchema(raw []byte) (gqlschem
 }
 
 func (p *servicePlanConverter) extractCreateSchema(raw []byte) (map[string]interface{}, error) {
+	raw = bytes.Trim(raw, "\x00")
 	extracted := make(map[string]interface{})
 
 	err := json.Unmarshal(raw, &extracted)
 	if err != nil {
 		return nil, errors.Wrap(err, "while extracting creation parameter schema")
 	}
+	if p.isEmptyCreateParameterSchema(extracted) {
+		return nil, nil
+	}
 
 	return extracted, nil
 }
+
+func (p *servicePlanConverter) isEmptyCreateParameterSchema(schema map[string]interface{}) bool {
```

Comment:
Maybe move it to a separate package within `internal` directory, as it is shared.

For example: `jsonschema.isEmpty()`

Reply:
I do not think it's a good idea, because in this case we check specific fields and the package you propose should be 100% generic.

Reply:
I can agree to extract this function + functions: `unpackCreateParameterSchema` and `extractCreateSchema` only to folder in this same domain (servicecatalog).

Reply:
@michal-hudy What do you thinking about this?

Reply:
@magicmatatjahu no, please, don't create new files in `servicecatalog` domain 😛 There are too many already.

If this is specific, then still, you can create subpackage like `servicecatalog/parameterschema` or `servicecatalog/jsonschema`